src/qilib/utils/storage/mongo.py

```diff
@@ -200,6 +203,8 @@ def _store_value_by_tag(self, tag: TagType, data: Any, parent: ObjectId,
               NodeAlreadyExistsError: If a tag in Tag List is an unexpected node/leaf
               NoDataAtKeyError:  If a tag in Tag List does not exist
         """
+        if self._read_only:
+            raise Exception('database {self._db.name} is read-only')
```

Comment:
It would be good to have a special class for this exception, similar to NodeAlreadyExistsError,  NoDataAtKeyError

Reply:
Done